### ISSUE TITLE
SCED code for courses

### ISSUE THREAD

**Issue by philbarker** [CredentialEngine/Schema-Development#798]
[SCED provides codes](https://nces.ed.gov/forum/SCED.asp) for secondary and pre-secondary courses, based on a combination of attributes: course code - course level - course credit or grade span [depending on whether the course is secondary or lower] - position in sequence. 
![sced](https://user-images.githubusercontent.com/658047/135468238-aa24688c-98f8-4188-89d0-9b74be15c4b5.png)

There's a "[finder](https://nces.ed.gov/scedfinder/Home/Search)" service, a [spreadsheet of subject codes](https://nces.ed.gov/forum/Current_SCED_File_Version_9.0.asp) and [a guide](https://nces.ed.gov/forum/pub_2014802.asp) (the image above is from the guide).

An example, from the SCED spreadsheet: 

> **Course title:** AP English Literature and Composition	
> **Course code:** 01006	
> **Course description:** Following the College Board’s suggested curriculum designed to parallel college-level English courses, AP English Literature and Composition courses enable students to develop critical standards for evaluating literature. Students study the language, character, action, and theme in works of recognized literary merit; enrich their understanding of connotation, metaphor, irony, syntax, and tone; and write compositions of their own (including literary analysis, exposition, argument, narrative, and creative writing).	

So if this is offered at Honors level for 5 Carnegie units as a stand-alone unit its full code would be 01001H5.0011

I don't think that we would want to disaggregate this code in order to store it -- just treat it as a string. Any system that understands SCED would be able to do that itself (though we might be able to parse it and use it to help provide information about the thing that is coded).

Though they call it a "course identifier" I don't think we would. Stuart called it a faceted classifier and I agree.   A SCEDs code is enough to uniquely identify a course as we think of it, i.e. one offered by a specific institution 

**Question:** is that how we think of a course? Am I right that we would think it wrong to see the same "course" offered at hundreds of high schools across the country, but instead you would expect each school to offer its own courses?

I see parallels with what SCED calls a "course" and Transfer Value Intermediaries #780  except, of course, that these aren't for transfer -- it identifies the group of courses that in this case all follow a certain curriculum and something similar must be true of transfer intermediaries, which is a form of classification.

Open questions (that I am following up with SCED)
- how widely used is SCED used
- where it is used, is the whole code set used or just the five-digit subject identifier code
- do we need to worry about the different versions of SCED (they're up to release 9).

Depending on these, we might want a new property to express SCED  or we might use an existing part of CTDL ([ceterms:InstructionalProgramClassification](https://purl.org/ctdl/terms/InstructionalProgramClassification) is a strong candidate).

**Comment by siuc-nate** [comment]
I don't think we need to overthink it.  Either we treat it as a classification (because it comes from a framework intended to group similar courses together) or we treat it as an identifier (because the institution uses it as an identifier and doesn't care that there are other courses in the same group) or both, and it's likely a string in any case as far as we're concerned.  We should keep this simple unless/until there is a reason to go deeper.

**Comment by philbarker** [comment]
@siuc-nate agreed, but we need to think about it enough to explain why we don't follow SCED in calling it a course identifier.

**Comment by siuc-nate** [comment]
From the perspective of the institutions that use it, it's an identifier - it doesn't matter (to them) that other courses at other institutions use the same one.  In other words, think of it like a bnode URI, it's a unique **identifier**, but it's only "unique" in the scope of its `@graph` (that is, unique within the institution's offerings). 

To use another, somewhat faulty analogy: There are many people named Phil, but only one Phil in our group, so we don't need to call out your entire name every time we want to identify you for our purposes.

So, if we want to use it as a classifier instead, our justification is basically just "our graph is bigger" since it has many such courses using the same "identifier" that conveniently also works as a classifier due to its faceted structure.  But then that also calls into question things like "English 101" - shouldn't any identifier like that also technically be a classifier? I don't think it's terribly useful to go there, so maybe the answer is we just use both properties - identifier and instructionalProgramType - since the same lexical string applies in two different ways.

To get really meta: You could argue that an identifier is just a classifier that happens to be specific enough to only apply to one thing (in a given scope).

**Comment by stuartasutton** [comment]
Phil, I don't think [ceterms:InstructionalProgramClassification](https://purl.org/ctdl/terms/InstructionalProgramClassification) is correct based on the clear distinction between course and program.  I don't think we need to do any "investigating" on use in PK-12. The SCED is used in U.S. government reporting. 

I think we should create a `ceterms:sced` property.

**Comment by siuc-nate** [comment]
Would the comment be a comment or a usage note?
